components/credentials/src/main/java/org/cloudfoundry/credhub/generators/CertificateGenerator.java

```diff
@@ -113,4 +128,15 @@ private boolean shouldUseTransitionalParentToSign(final Boolean allowTransitiona
 
     return transitionalVersionCreatedAt.isAfter(latestNonTransitionalVersionCreatedAt);
   }
+
+  private int getCertificateDuration(final CertificateGenerationParameters params) {
+    if (params.isCa()) {
+      return Math.max(params.getDuration(), caMinimumDurationInDays);
+    }
+    return Math.max(params.getDuration(), leafMinimumDurationInDays);
+  }
+
+  private boolean isMinimumDurationEnabled() {
```

Comment:
That actually appears to be logically incorrect, isn't it? 

If the CA minimum duration is set to 3, and the leaf isn't set, and you are generating a leaf, this function will return `true`, which will result in the `durationOverwritten` field to be `true` or `false`, as if the feature was enabled for leaves. While it was only enabled for CAs.

So I didn't realize during my first review that it was important to know if the feature was enabled or not, for the purpose of showing the information in the response or not.

So it's annoying. It looks like you need to have 2 functions: `isMinimumDurationEnabledForCAs` and `...ForLeaves`.

Unless you actually assume that if a user sets one of them, then the whole feature is enabled. I guess that that's what you had in mind.

Reply:
> Unless you actually assume that if a user sets one of them, then the whole feature is enabled. I guess that that's what you had in mind.

Yes. We considered enabling either a leaf minimum or a CA minimum to be enabling the feature as a whole. However, since we've removed the null response, this method has been removed entirely.

Reply:
👌 

We may want this function back if we want to be able to disable the feature, and mentioned in a previous conversation.